### PR TITLE
Add Doxygen support

### PR DESCRIPTION
**Pull request by drizt**

### CODE REVIEW

**Comment by drizt** [comment]
Use `make doc` to build documentation. then `xdg-open doc/html/index.html` to open.

**Comment by kuba--** [comment]
If we wanna support _doxygen_ we should deliver default config files as well. Right now, you have to create your own `Doxyfile`.

**Comment by drizt** [comment]
Forget to commit.

File: Doxyfile.in [hunk 1]
```diff
@@ -0,0 +1,2482 @@
...
+
+DOXYFILE_ENCODING      = UTF-8
+
+# The PROJECT_NAME tag is a single word (or a sequence of words surrounded by
+# double-quotes, unless you are using Doxywizard) that should identify the
+# project for which the documentation is generated. This name is used in the
+# title of most generated pages and in a few other places.
+# The default value is: My Project.
+
+PROJECT_NAME           = "kuba zip"
```
[the remark this answers is not in the document]
Comment (drizt) [reply Doxyfile.in]: Changed

File: Doxyfile.in [hunk 2]
```diff
@@ -0,0 +1,2482 @@
...
+
+# The PROJECT_NUMBER tag can be used to enter a project or revision number. This
+# could be handy for archiving the generated documentation or if some version
+# control system is used.
+
+PROJECT_NUMBER         =
```
[the remark this answers is not in the document]
Comment (drizt) [reply Doxyfile.in]: changed

**Comment by drizt** [comment]
Fixed.

File: CMakeLists.txt [hunk 3]
```diff
@@ -3,6 +3,8 @@ project(zip)
 enable_language(C)
 set(CMAKE_MODULE_PATH "${PROJECT_SOURCE_DIR}/cmake" ${CMAKE_MODULE_PATH})
 
+set(ZIP_VERSION "0.1.9")
```
Comment (kuba--) [inline CMakeLists.txt]: why 0.1.9?
the latest version is `v0.1.14`, so it can go to the candidate `v0.1.15`

Comment (drizt) [reply CMakeLists.txt]: I took version from appveyor.yml.

Comment (drizt) [reply CMakeLists.txt]: Changed to 0.1.15.

Comment (kuba--) [reply CMakeLists.txt]: ok, so please push changes

Comment (drizt) [reply CMakeLists.txt]: Done.

**Pull request merged**